src/ctap2/commands/client_pin.rs

```diff
@@ -272,6 +321,78 @@ impl<'sc, 'pin> ClientPINSubCommand for GetPinToken<'sc, 'pin> {
     }
 }
 
+#[derive(Debug)]
+pub struct GetPinUvAuthTokenUsingPinWithPermissions<'sc, 'pin> {
+    pin_protocol: u8,
+    shared_secret: &'sc ECDHSecret,
+    pin: &'pin Pin,
+    permissions: PinUvAuthTokenPermission,
+    rp_id: Option<String>,
+}
+
+impl<'sc, 'pin> GetPinUvAuthTokenUsingPinWithPermissions<'sc, 'pin> {
+    pub fn new(
+        info: &AuthenticatorInfo,
+        shared_secret: &'sc ECDHSecret,
+        pin: &'pin Pin,
+        permissions: PinUvAuthTokenPermission,
+        rp_id: Option<String>,
+    ) -> Result<Self, CommandError> {
+        // TODO(MS)!
```

Comment:
What's the TODO here?

Reply:
To actually handle protocol 2. I'll update the comment.